vendor/github.com/openshift/library-go/pkg/operator/resource/resourcemerge/generic_config_merger.go

```diff
@@ -40,11 +43,10 @@ func MergeProcessConfig(specialCases map[string]MergeFunc, configYAMLs ...[]byte
 			// maybe it's just json
 			prevConfigJSON = currentConfigYAML
 		}
-		prevConfigObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, prevConfigJSON)
-		if err != nil {
+		prevConfig := map[string]interface{}{}
+		if err := json.NewDecoder(bytes.NewBuffer(prevConfigJSON)).Decode(&prevConfig); err != nil {
```

Comment:
What was wrong with previous decode into unstructured? Is this direct json encoding better/simpler/more performant?

Reply:
> What was wrong with previous decode into unstructured? Is this direct json encoding better/simpler/more performant?

Decoding into unstructured requires a GVK be specified and a map doesn't